src/omv/imlib/simd.h

```diff
@@ -129,32 +143,85 @@ static inline v128_predicate_t vpredicate_8(uint32_t n) {
     #if (__ARM_ARCH >= 8)
     return vctp8q(n);
     #else
-    return IM_MIN(n, UINT8_VECTOR_SIZE);
+    return (n < UINT8_VECTOR_SIZE) ? n : UINT8_VECTOR_SIZE;
+    #endif
+}
+
+// Use this for code that has a scalar fallback path to automatically remove predication
+// on architectures with small vector sizes where it can hurt performance...
+static inline uint32_t vpredicate_8_maybe(uint32_t n) {
+    #if (VECTOR_SIZE_BYTES > 4)
+    return vpredicate_8(n);
+    #else
+    (void) n;
+    return 4;
+    #endif
+}
+
+static inline uint32_t vpredicate_8_maybe_min_elements() {
+    #if (VECTOR_SIZE_BYTES > 4)
+    return 1;
+    #else
+    return 4;
     #endif
 }
 
 static inline v128_predicate_t vpredicate_16(uint32_t n) {
     #if (__ARM_ARCH >= 8)
     return vctp16q(n);
     #else
-    return IM_MIN(n, UINT16_VECTOR_SIZE);
+    return (n < UINT16_VECTOR_SIZE) ? n : UINT16_VECTOR_SIZE;
+    #endif
+}
+
+// Use this for code that has a scalar fallback path to automatically remove predication
+// on architectures with small vector sizes where it can hurt performance...
+static inline uint32_t vpredicate_16_maybe(uint32_t n) {
+    #if (VECTOR_SIZE_BYTES > 4)
+    return vpredicate_16(n);
+    #else
+    (void) n;
+    return 2;
+    #endif
+}
+
+static inline uint32_t vpredicate_16_maybe_min_elements() {
+    #if (VECTOR_SIZE_BYTES > 4)
+    return 1;
+    #else
+    return 2;
```

Comment:
For cases like these, you can just write:
```C
return (VECTOR_SIZE_BYTES > 4) ? 1 : 2;
```
It should be reduced to a constant.

Reply:
Done